From 5cb572d7516cd8c00301d82917122d3677f9701b Mon Sep 17 00:00:00 2001
From: Gabriel Wicki <gabriel.wicki@bytesatwork.ch>
Date: Tue, 14 Jun 2022 11:55:08 +0200
Subject: [PATCH] bytedevkit: Use `hwid check` to select dtb

`hwid check` sets the board_* variables. These are in turn used to set the
`dtbfile` variable.

Signed-off-by: Gabriel Wicki <gabriel.wicki@bytesatwork.ch>
---
 include/configs/bytedevkit.h | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/include/configs/bytedevkit.h b/include/configs/bytedevkit.h
index 753ccba02b..249abc6471 100644
--- a/include/configs/bytedevkit.h
+++ b/include/configs/bytedevkit.h
@@ -80,6 +80,7 @@
 #if !defined(CONFIG_SPL_BUILD)
 
 #define CONFIG_BOOTCOMMAND \
+	"run select_dtb; " \
 	"mmc rescan; " \
 	"run mmc_boot; " \
 
@@ -97,7 +98,6 @@
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"mmc_dev=0:4\0" \
 	"mmc_root=/dev/mmcblk0p4\0" \
-	"dtbfile=/boot/stm32mp157c-bytedevkit.dtb\0" \
 	"kernelfile=/boot/uImage\0" \
 	"loadaddr=0xc1000000\0" \
 	"stdin=serial\0" \
@@ -122,6 +122,15 @@
 	"default_args=rootwait rw vt.global_cursor_default=0 consoleblank=0\0" \
 	"mmc_args=setenv bootargs ${default_args} console=${console} " \
 		"root=${mmc_root} ${bootargs_append}; \0" \
+	"select_dtb=if hwid check; " \
+		"then " \
+			"setenv dtbfile \"" \
+			"/boot/stm32mp157c-bytedevkit-" \
+			"v${board_major}-${board_minor}.dtb\"; " \
+		"else " \
+			"setenv dtbfile " \
+			"\"/boot/stm32mp157c-bytedevkit-v1-1.dtb\";" \
+		"fi;\0" \
 	"update_spiflash=echo Updating SPI Flash ...; " \
 		"sf probe 0; " \
 		"sf erase 0 +${spl_uboot_size}; " \
-- 
2.20.1

